compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6256,17 +6256,7 @@ private void checkStringTemplateExprs(List<? extends BLangExpression> exprs, boo
                 continue;
             }
 
-            if (type.tag >= TypeTags.JSON) {
-                if (allowXml) {
-                    if (type.tag != TypeTags.XML) {
-                        dlog.error(expr.pos, DiagnosticErrorCode.INCOMPATIBLE_TYPES,
-                                BUnionType.create(null, symTable.intType, symTable.floatType,
-                                        symTable.decimalType, symTable.stringType,
-                                        symTable.booleanType, symTable.xmlType), type);
-                    }
-                    continue;
-                }
-
+            if (!types.isUnionOfSimpleBasicTypesExceptNilType(type)) {
```

Comment:
```suggestion
            if (!types.isNonNilSimpleBasicTypeOrString(type)) {
```

Reply:
Cant, we use assignability here?

Reply:
Then we'll have to have a separate union type for this, right?